src/server/bitops_family_test.cc

```diff
@@ -22,6 +28,159 @@ using absl::StrCat;
 namespace dfly {
 
+class Bytes {
+  using char_t = std::uint8_t;
+  using string_type = std::basic_string<char_t>;
+
+ public:
+  enum State { GOOD, ERROR, NIL };
+
+  Bytes(std::initializer_list<std::uint8_t> bytes) : data_(bytes.size(), 0) {
+    // note - we want this to be like its would be used in redis where most significate bit is to
+    // the "left"
+    std::copy(rbegin(bytes), rend(bytes), data_.begin());
+  }
+
+  explicit Bytes(unsigned long long n) : data_(sizeof(n), 0) {
+    FromNumber(n);
+  }
+
+  static Bytes From(unsigned long long x) {
+    return Bytes(x);
+  }
+
+  explicit Bytes(State state) : state_{state} {
+  }
+
+  Bytes(const char_t* ch, std::size_t len) : data_(ch, len) {
+  }
+
+  Bytes(const char* ch, std::size_t len) : Bytes(reinterpret_cast<const char_t*>(ch), len) {
+  }
+
+  explicit Bytes(std::string_view from) : Bytes(from.data(), from.size()) {
+  }
+
+  static Bytes From(RespExpr&& r) {
+    if (r.type == RespExpr::STRING) {
+      return Bytes(ToSV(r.GetBuf()));
+    } else {
+      if (r.type == RespExpr::NIL || r.type == RespExpr::NIL_ARRAY) {
+        return Bytes{Bytes::NIL};
+      } else {
+        return Bytes(Bytes::ERROR);
+      }
+    }
+  }
+
+  std::size_t size() const {
+    return data_.size();
+  }
+
+  operator std::string_view() const {
+    return std::string_view(reinterpret_cast<const char*>(data_.data()), size());
+  }
+
+  std::ostream& print(std::ostream& os) const {
+    if (state_ == GOOD) {
+      for (auto c : data_) {
+        std::bitset<8> b{c};
+        os << b << ":";
+      }
+    } else {
+      if (state_ == NIL) {
+        os << "nil";
+      } else {
+        os << "error";
+      }
+    }
+    return os;
+  }
+  std::ostream& print_hex(std::ostream& os) const {
+    if (state_ == GOOD) {
+      for (auto c : data_) {
+        os << std::hex << std::setfill('0') << std::setw(2) << (std::uint16_t)c << ":";
+      }
+    } else {
+      if (state_ == NIL) {
+        os << "nil";
+      } else {
+        os << "error";
+      }
+    }
+    return os;
+  }
+
+ private:
+  template <typename T> void FromNumber(T num) {
+    // note - we want this to be like its would be used in redis where most significate bit is to
+    // the "left"
+    std::size_t i = 0;
+    for (const char_t* s = reinterpret_cast<const char_t*>(&num); i < sizeof(T); s++, i++) {
+      data_[i] = *s;
+    }
+  }
+
+  string_type data_;
+  State state_ = GOOD;
+};
+
+inline bool operator==(const Bytes& left, const Bytes& right) {
+  return static_cast<const std::string_view&>(left) == static_cast<const std::string_view&>(right);
+}
+
+inline bool operator!=(const Bytes& left, const Bytes& right) {
+  return !(left == right);
+}
+
+inline Bytes operator"" _b(unsigned long long x) {
+  return Bytes::From(x);
+}
+
+inline Bytes operator"" _b(const char* x, std::size_t s) {
+  return Bytes{x, s};
+}
+
+inline Bytes operator"" _b(const char* x) {
+  return Bytes{x, std::strlen(x)};
+}
+
+class hexprint {
```

Comment:
why do you need this class? why not do it directly?

Reply:
removed